pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -290,6 +290,7 @@ func (r *MemberRollReconciler) reconcileObject(ctx context.Context, roll *maistr
 	}
 
 	// 5. check each ServiceMeshMember object to see if it's configured or terminating
+	allKnownMembers := sets.NewString(roll.Spec.Members...).Insert(getNamespaces(members)...).Delete(meshNamespace)
```

Comment:
I think we should also filter out namespaces that do not exist.  Personally, I don't think we should allow configuration of members that don't exist.  Maybe we can leave this as is, and move the discussion to member roll validation (I think there's an existing issue).

Reply:
can't we leave that validation to kiali?

Reply:
Based on my tests, Kiali seems to handle missing namespaces by itself. 

I do agree that we should probably not allow users to add nonexistent namespaces to the SMMR. We don't even need to add any kind of validation to the SMMR. We can simply remove the `.spec.members` field. Actually, adding validation wouldn't even work properly, as users could delete the namespace and we'd end up with an SMMR pointing to a nonexistent namespace again. 

Mesh admins can create a member namespace, create a ServiceMeshMember resource inside it, and then grant admin access to the namespace to a particular mesh user. This way, there's no way of somebody else gaining access to the mesh simply by creating the namespace that they know is specified in the SMMR.